libcst/codemod/commands/rename.py

```diff
@@ -0,0 +1,192 @@
+from typing import Optional, Sequence, Union
+
+import libcst as cst
+from libcst.codemod import CodemodContext, VisitorBasedCodemodCommand
+from libcst.codemod.visitors import AddImportsVisitor
+from libcst.helpers import get_full_name_for_node
+from libcst.metadata import QualifiedName, QualifiedNameProvider, QualifiedNameSource
+
+
+class RenameCommand(VisitorBasedCodemodCommand):
+    """
+    Rename all instances of a variable or import.
+    """
+
+    DESCRIPTION: str = (
+        "Rename all instances of a variable or import. "
+        + "The existing import may not be removed and RenameCommand needs more work to be perfect."
+    )
+
+    METADATA_DEPENDENCIES = (QualifiedNameProvider,)
+
+    @staticmethod
+    def add_args(parser: argparse.ArgumentParser) -> None:
+        parser.add_argument(
+            "--orig_module",
+            dest="orig_module",
+            help="Name of the module to rename. Leave blank if renaming a local variable.",
+        )
+        parser.add_argument(
+            "--orig_object",
+            dest="orig_object",
+            required=True,
+            help="Name of the object in module or variable to rename.",
+        )
+        parser.add_argument(
+            "--new_module",
+            dest="new_module",
+            help="New name of the module. Leave blank if renaming a local variable, or if same as orig_module.",
+        )
+        parser.add_argument(
+            "--new_object",
+            dest="new_object",
+            required=True,
+            help="New name of the object in module or variable.",
+        )
+
+    def __init__(
+        self,
+        context: CodemodContext,
+        orig_object: str,
+        new_object: str,
+        orig_module: Optional[str] = None,
+        new_module: Optional[str] = None,
+    ) -> None:
+        super().__init__(context)
+
+        self.orig_module: Optional[str] = orig_module
+        self.orig_object: str = orig_object
+        self.orig_name: str = (
+            f"{orig_module}.{orig_object}" if orig_module is not None else orig_object
+        )
+
+        self.new_module: Optional[
+            str
+        ] = new_module if new_module is not None else orig_module
+        self.new_object: str = new_object
+        self.new_name: str = (
+            f"{new_module}.{new_object}" if new_module is not None else new_object
+        )
+
+        self.source: QualifiedNameSource = (
+            QualifiedNameSource.IMPORT
+            if orig_module is not None
+            else QualifiedNameSource.LOCAL
+        )
+
+    def leave_Import(
+        self, original_node: cst.Import, updated_node: cst.Import
+    ) -> Union[cst.Import, cst.RemovalSentinel]:
+        for import_alias in original_node.names:
+            name = get_full_name_for_node(import_alias.name)
+            if name is not None and name == self.orig_module:
+                # Remove old import altogether
+                return cst.RemoveFromParent()
```

Comment:
In the case of `import a, b, c`, there will be multiple `names`.
If only `b` is the target to be removed, this will remove the entire import statement which is wrong.
We should just create a new list of `names` by filtering the existing names and then return the new `names` in an updated `Import` node.
Let's also add test cases to cover this case.

Reply:
We do not want to use an import if it is referenced in code (The internal version of rename handles this I believe). A `from` import can also be used to import a module which is used in multiple places (so solution 1 is not a viable solution).

Reply:
Yeah this was on my mind as well. But then I realized that the way arguments are passed to the codemod, the imported module won't get removed. And this is checked in the test [test_no_removal_of_import_from_in_use](https://github.com/Instagram/LibCST/pull/308/files#diff-511afb8b206c354e493c782e1c50abadR245)